tests/components/zha/test_sensor.py

```diff
@@ -300,6 +300,19 @@ async def async_test_device_temperature(hass: HomeAssistant, cluster, entity_id)
     assert_state(hass, entity_id, "29.0", UnitOfTemperature.CELSIUS)
 
 
+async def async_test_setpoint_change_source(hass, cluster, entity_id):
+    """Test the translation of numerical state into enum text."""
+    await send_attributes_report(hass, cluster, {0x0030: 0x01})
+    hass_state = hass.states.get(entity_id)
+    assert hass_state.state == "Schedule"
+
+
+async def async_test_pi_heating_demand(hass, cluster, entity_id):
+    """Test pi heating demand is correctly returned."""
+    await send_attributes_report(hass, cluster, {0x0008: 1})
```

Comment:
`Thermostat.AttributeDefs.pi_heating_demand.id` instead of `0x0008`

Reply:
Done